Add Vector class

NoahRoseLedesma opened the pull request:
Add a general 2D vector class. Has methods for all the vector math you will need.

Commit:
Add Vector class

adamperlin commented on Vector2D.java:
```diff
@@ -0,0 +1,66 @@
+    {
+        return Math.sqrt( ( Math.pow(m_nXComponent, 2 ) ) + ( Math.pow(m_nYComponent, 2 ) ) );
+    }
+    public double GetDirection()
```
This should be `Math.atan2` instead of `Math.atan`. `Math.atan` returns angles from `[-pi/2 pi/2]`, but `Math.atan2` returns angles from `[-pi, pi]`, and takes both the `x` and `y` components of the vector as arguments. If the vector was `<1, 1>`, the function in its current state would return `pi/4`. If it was `<-1, -1>`, it would also return `pi/4`, when it should return `-3pi/4` (or `5pi/4`)

adamperlin commented on Vector2D.java:
```diff
@@ -0,0 +1,66 @@
+    private double m_nYComponent;
+
+
+    public Vector2D( double iXComponent, double iYComponent ) {}
```
Maybe set fields in this constructor, so the vector is initialized.

NoahRoseLedesma replied:
Lol whoops.

adamperlin commented on Vector2D.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    public void Set( Vector2D vNewVector ) { this.SetComponents( vNewVector.GetXComponent(), vNewVector.GetYComponent() ); }
+    public void Set( double nMagnitude, double nDirection )
```
Maybe change name to `setPolar` or something to indicate that it takes polar coordinates.

adamperlin commented on Vector2D.java:
```diff
@@ -0,0 +1,66 @@
+    {
+        return ( m_nXComponent * vVector.GetXComponent() ) + ( m_nYComponent * vVector.GetYComponent() );
+    }
+}
```
Maybe add a `multiplyScalar` method that scales both components the same amount.

adamperlin reviewed (changes requested):
Looks good, there are just a few small changes that would make this more useable. The biggest issue is the atan bug.

Commit:
Fix small errors in Vector.java

Updates based on feedback from Adam in #3. Thanks :+1:
 - Fix broken constructor, overload constructor.
 - Use Math.atan2 instead of Math.atan ( See #3 for reasoning )
 - Setting, Adding, and Multiplying can be performed with polar form vectors using SetPolar, AddPolar, and MultiplyPolar respectively.
 - Add MultiplyScalar method.

Commit:
Correct arguments for Math.atan2

Was passing the Y-Component divided by the X-Component but math.atan2 takes both components as arguments.

adamperlin commented:
Thanks for the changes, Noah :+1:
